src/ucp/core/ucp_mm.c

```diff
@@ -727,13 +733,7 @@ static void ucp_memh_init(ucp_mem_h memh, ucp_context_h context,
     memh->alloc_method   = method;
     memh->mem_type       = mem_type;
     memh->sys_dev        = sys_dev;
-
-    /* Cache sys_dev in a format packed to rkey to minimize overhead during
-     * rndv protocols. TODO remove if using another method to mark rkey with
-     * remote flush requirement. */
-    memh->packed_sys_dev = (sys_dev == UCS_SYS_DEVICE_ID_UNKNOWN) ?
-                                   UCS_SYS_DEVICE_ID_UNKNOWN :
-                                   ucp_rkey_pack_sys_dev(memh);
+    memh->packed_sys_dev = sys_dev;
```

Comment:
maybe instead of saving "memh->packed_sys_dev" we add FLUSH flag to memh->flags (we set it during registration if on the the MDs we register used a sibling)?